### PR TITLE
Added functionality for keys in the config json, updated bids compatible run id, updated Dockerfile

### PR DESCRIPTION
**Pull request by slimnsour**
This pull request is for the initial changes we want for our own dcm2bids usage. For each change, the relevant documentation and tests have been updated.

- You can now use keys instead of integer indices for use in the intendedFor field. See `docs/3-configuration.md` for more info, as well as the new example config file at `example/key_config.json`.
- Changed the run id parsing to remove unnecessary 0 to be more compliant with bids + fmriprep. For example, `run-01` becomes `run-1`. 
- Updated the `Dockerfile` and `setup.py` to be more like those in our other projects (tractify and dmri_analyses).

Feedback is greatly appreciated! We'll be able to add more features using this repo as well.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Use Ubuntu 20.04 LTS
+FROM ubuntu:latest
```
Comment (josephmje) [inline Dockerfile]: Specify container version here rather than latest

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+                    ca-certificates \
+                    curl \
+                    unzip \
+                    git
+
+WORKDIR /src/dcm2niix_build
+
+RUN curl -sSLO https://github.com/rordenlab/dcm2niix/releases/latest/download/dcm2niix_lnx.zip && \
```
Comment (josephmje) [inline Dockerfile]: Specify latest version explicitly.

File: dcm2bids/structure.py [hunk 3]
```diff
@@ -217,27 +217,47 @@ def intendedFor(self, value):
...
         """
         """
         data = self.srcSidecar.origData
         data["Dcm2bidsVersion"] = __version__
 
         # intendedFor key
+        # Check if intendedFor is None
         if self.intendedFor != [None]:
             intendedValue = []
             for index in self.intendedFor:
```
Comment (josephmje) [inline dcm2bids/structure.py]: Maybe use another variable name here since it can be either an index or string.

EDIT: Comment is for line 230

**Review by josephmje — COMMENTED** [review]
Thanks for this @slimnsour! I just left a few minor comments

File: Dockerfile (hunk 1 above)
Comment (slimnsour) [reply Dockerfile]: Thanks for the comments! Addressed in 3f0793d45358bfb6aa311f721159a923c2af6674

File: Dockerfile (hunk 2 above)
Comment (slimnsour) [reply Dockerfile]: Addressed in 3f0793d45358bfb6aa311f721159a923c2af6674

File: dcm2bids/structure.py (hunk 3 above)
Comment (slimnsour) [reply dcm2bids/structure.py]: Addressed in 3f0793d45358bfb6aa311f721159a923c2af6674

**Review by josephmje — APPROVED** [review]

**Pull request merged**